app/javascript/packages/session/requests.ts

```diff
@@ -37,7 +37,9 @@ const mapSessionStatusResponse = ({ live, timeout }: SessionStatusResponse): Ses
  * @return A promise resolving to the current session status
  */
 export const requestSessionStatus = (): Promise<SessionStatus> =>
-  request<SessionStatusResponse>(SESSIONS_URL).then(mapSessionStatusResponse);
+  request<SessionStatusResponse>(SESSIONS_URL)
+    .catch(() => ({ live: false, timeout: new Date().toISOString() }))
+    .then(mapSessionStatusResponse);
```

Comment:
I should backport this (b9aeba1479932a7fd0cddff33024c5a5503e5bfe) to #7966. It was necessary to fix the specs here, but is also something a real user may encounter. As mentioned in the commit description, Warden will set a response status code of 401 if the request includes a cookie for an expired session, which is likely to occur in the real-world.

Separately, it might be an improvement to make `timeout` and optional or nullable key in the response, since it'd avoid the awkwardness of putting "now", especially after #7966's a9a2bb1e8d611f976b3a62033a5a2fd7b4079891 which considers "now" as valid for `live?`.